### ISSUE TITLE
ＤＥＣ method does not work under current version (2017.1.20) of mxnet 

### ISSUE THREAD

**Issue by XifengGuo** [apache/incubator-mxnet#4763]
I run "python mxnet/example/dec/dec.py" under version 0.7 of mxnet and can get good result of clustering accuracy 84%.
But when I updated mxnet to the newest version, running the same code only gets accuracy of about 24%.
Then I reverse back to version 0.7, the accuracy can reach 84% again.
So it should be that something is changed in the new version, resulting the failure of running dec.py. But I can not figure it out. Did someone try this method and encounter the same issue?

## Environment info
Operating System:  Ubuntu 16.04
Compiler:  python 2.7
Package used (Python/R/Scala/Julia):
           scikit-learn
MXNet version: 
Installed from source:
MXNet commit hash (`git rev-parse HEAD`): f2c2b52a4a00ac5c99604ecfd170066124359fe0

If you are using python package, please provide

Python version and distribution:
  python 2.7

## Error Message:
No errors, but the result of running 'example/dec/dec.py' is very very bad using the newest version of mxnet.

**Comment by harusametime** [comment]
I got accuracy of 12% by the latest version, and saw the result that all the MNIST data were classified into one cluster.
Probably, layer-wise pretraining in autoencoder does not work well because pretraining layer 2 & 3 always  outputs zero L2-loss as metric.

Thank you for your helpful information, XifengGuo.
I also ran version 0.7 by
`git clone --recursive --depth=1 -b v0.7.0 https://github.com/dmlc/mxnet.git`
and get accuracy of 82 %.

**Comment by vwrobel** [comment]
Hi,
I wanted to try the DEC MXNET implementation. I have installed the 0.7 version as you suggested.
I am kind of stuck with a clustering accuracy of 71% on the MNIST dataset. Did you modify the example parameters to reach 82%?
Cheers.

**Comment by hukangrong** [comment]
@harusametime also can't work. the net's parameter not changed.

**Comment by hukangrong** [comment]
`help(mxnet)
Help on package mxnet:`
```
VERSION
    0.7.0
```

**Comment by harusametime** [comment]
@hukangrong 
What is wrong in your case? Is the stacked autoencoder trained well?

**Comment by hukangrong** [comment]
@harusametime I think the problem is happened in  0.9-version's "mnist_pt.arg" file, tha  I used it to run 0.7 version's code.

**Comment by szha** [comment]
This issue is closed due to lack of activity in the last 90 days. Feel free to ping me to reopen if this is still an active issue. Thanks!
